README.md

```diff
@@ -1,16 +1,180 @@
+  - [`inspect()`](#inspect)
+  - [`release()`](#release)
+- [Events](#events)
+  - [`'acquired'`](#acquired)
+  - [`'refreshed'`](#refreshed)
+  - [`'released'`](#released)
+  - [`'error'`](#error)
+- [Authors](#authors)
+- [License](#license)
+- [Contributing](#contributing)
+
+<!-- /TOC -->
+## `ExclusiveLock`
+
+Need to synchronize a bunch of workers so that only 1 does a particular thing?  If yes,
+then you need a lock file.  This package uses a cache store (`redis` or `keydb`)
+to create records atomically that only 1 instance will "own."
+
+### NOTES
+
+This library covers most common use cases for exclusion when working with multiple instances of the same application / container:
+
+* Exclusive initialization: have one of the application instances perform a task on init.
+* Periodic checks: Have one of the application instances perform a task periodically.
+
+For long running tasks and rare edge cases, it could be possible that the lock is perceived as acquired by more than one instance. This library does not provide strict guarantees for exclusion for tasks that must be processed exactly once in a non-idempotent manner (e.g. appending an item in order processing). Consider using upserts/cas on your persistent DB in those cases or using a dedicated solution for exclusion like etcd / zookeeper.
 
 ## Installation
 
+```bash
+npm install @logdna/exclusive-lock --save
+```
 
 ## Usage
 
+```js
+const ExclusiveLock = require('@logdna/exclusive-lock')
+const cache_connection = require('./my-keydb-connection.js')
+const pino = require('pino')
+
+const exclusive_lock = new ExclusiveLock({
+  name: 'my-distributed-app'
+, log: pino()
+, cache_manager
+})
+
+async function main() {
+  const acquired = await exclusive_lock.acquire()
+  if (acquired) {
+    log.info('You win the race! Lock acquired')
+  }
+  exclusive_lock.release()
+}
+
+main()
+```
+
+## Auto-refreshing
+As a feature, `exclusive-lock` will start a timer to automatically refresh the TTL on
+the lock file, as it's assumed that the lock holder will always want to hold the lock
+until explicitly released. This feature is handled by a `setInterval` timer and can
+encouter errors asynchronously.  See the [`error`](#error) event for details on that.
+
+* The timer is not started until a lock is successfully [`acquired`](#acquire).
+* For clean shutdown, always [`release()`](#release) an acquired lock so that the timer is canceled.
+* [`refreshed`](#refreshed) is emitted for every successful refresh.
+
+## API
+
+-------
+
+### `new ExclusiveLock(options)`
+* `options` [`<Object>`][]
+  * `name` [`<String>`][] - **Required** A unique string to identify your application
+  * `cache_connection` [`<Object>`][] - **Required** A connection to the cache, either Redis or Keydb
+  * `log` [`<Object>`][] - A optional logger instance such as `pino`. Must support levels `info`, `warn`, `error`, `debug`. **Default: [abstract-logging](https://www.npmjs.com/package/abstract-logging)**
```

Comment:
```suggestion
  * `log` [`<Object>`][] - A optional logger instance such as `pino`. Must support levels `info`, `warn`, `error`, `debug` and the `child` method. **Default: [abstract-logging](https://www.npmjs.com/package/abstract-logging)**
```
```suggestion
  * `log` [`<Object>`][] - A optional logger instance such as `pino`. Must support levels `info`, `warn`, `error`, `debug`. **Default: [abstract-logging](https://www.npmjs.com/package/abstract-logging)**
```

Reply:
I don't understand the need for a `child`.  There is nothing that is making `child` off of the incoming logger.  The only way this would error is if someone tries to do `cache_connection.log.child()`, which wouldn't make sense because the caller has access to the original logger.